biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java

```diff
@@ -448,6 +579,48 @@ public Manifest getManifest() throws Exception {
 		return manifest().orElse(null);
 	}
 
+	/**
+	 * Creates a <b>copy</b> of the current jars manifest that is enhanced by
+	 * the supplemental manifest data (if any) for the given release.
+	 *
+	 * @param release the release for fetching an enhanced manifest
+	 * @return a <b>copy</b> that is <b>not</b> backed by the original manifest
+	 *         but copied and enhanced with the supplemental manifest data (if
+	 *         any) for the given release
+	 */
+	public Optional<Manifest> getManifest(int release) {
+		if (isMultiRelease()) {
+			return manifest().map(original -> {
+				Manifest copy = new Manifest(original);
+				if (release >= MULTI_RELEASE_MIN_VERSION) {
+					Optional<Resource> releaseEntry = getVersionedResource(JarFile.MANIFEST_NAME, release);
```

Comment:
Having checked the specification the location of supplemental manifests is `/META-INF/versions/X/OSGI-INF/MANIFEST.MF`. See https://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#framework.module-multireleasejar